packages/db-collections/src/electric.ts

```diff
@@ -77,33 +77,179 @@ export interface ElectricCollectionConfig<
 
   /**
    * Optional asynchronous handler function called before an insert operation
-   * Must return an object containing a txid string
+   * Must return an object containing a txid string or array of txids
    * @param params Object containing transaction and mutation information
-   * @returns Promise resolving to an object with txid
+   * @returns Promise resolving to an object with txid or txids
+   * @example
+   * // Basic Electric insert handler - MUST return { txid: string }
+   * onInsert: async ({ transaction, collection }) => {
+   *   const newItem = transaction.mutations[0].modified
+   *   const result = await electric.db.todos.create({
+   *     data: newItem
+   *   })
+   *   return { txid: result.txid } // Required for Electric sync matching
+   * }
+   *
+   * @example
+   * // Insert handler with multiple items - return array of txids
+   * onInsert: async ({ transaction, collection }) => {
+   *   const items = transaction.mutations.map(m => m.modified)
+   *   const results = await Promise.all(
+   *     items.map(item => electric.db.todos.create({ data: item }))
+   *   )
+   *   return { txid: results.map(r => r.txid) } // Array of txids
+   * }
+   *
+   * @example
+   * // Insert handler with batch operation - single txid
+   * onInsert: async ({ transaction, collection }) => {
+   *   const items = transaction.mutations.map(m => m.modified)
+   *   const result = await electric.db.todos.createMany({
+   *     data: items
+   *   })
+   *   return { txid: result.txid } // Single txid for batch operation
+   * }
+   *
+   * @example
+   * // Insert handler with await for sync confirmation
+   * onInsert: async ({ transaction, collection }) => {
+   *   const newItem = transaction.mutations[0].modified
+   *   const result = await electric.db.todos.create({
+   *     data: newItem
+   *   })
+   *
+   *   // Wait for the txid to be synced from server
+   *   await collection.utils.awaitTxId(result.txid)
+   *
+   *   return { txid: result.txid }
+   * }
    */
   onInsert?: (
     params: InsertMutationFnParams<ResolveType<TExplicit, TSchema, TFallback>>
-  ) => Promise<{ txid: string }>
+  ) => Promise<{ txid: string | Array<string> }>
```

Comment:
I realized writing the docs that we should also support returning an array of txids e.g. if you do multiple api calls